lib/autoupdate/agent/telemetry.go

```diff
@@ -0,0 +1,61 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package agent
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/gravitational/trace"
+)
+
+// IsActive returns true if the local Teleport binary is managed by teleport-update.
+// Note that true may be returned even if auto-updates is disabled or the version is pinned.
+func IsActive() (bool, error) {
+	teleportPath, err := os.Readlink("/proc/self/exe")
+	if err != nil {
+		return false, trace.Wrap(err, "cannot find Teleport binary")
+	}
+	updaterBasePath := filepath.Clean(teleportOptDir) + "/"
```

Comment:
Why the Clean on a constant path?

Reply:
I always clean path comparisons first (`Join` does this automatically). Otherwise, you can end up comparing different representations of the same path. The idiomatic way to do this is to `os.Stat` and `os.SameFile`, but this code compares prefixes, and I didn't want to make assumptions about the placement of `teleport` within `/opt/teleport/[install]/`

I realize this feels a bit silly on a constant, but I'm trying to be very defensive against future changes, since the updater runs on infrastructure that we can't debug as easily (at least from Cloud's perspective). (It's also unnecessary to call `Abs` on `/proc/self/exe` values.)

I'll extract and test a method to better cover / document the utility of these functions.

Reply:
Maybe clean/abs only the very last parts, before any comparisons? The Clean and Abs in contexts where they do nothing send a strange signal to readers of the code, as it makes me wonder what the heck I am missing about it.

Reply:
Let me know if the new structure looks better

Reply:
It's great, thanks. The new comments are very helpful too.